File: smartfreelance-backend/project-service/src/main/java/com/smartfreelance/projectservice/service/TaskService.java
package com.smartfreelance.projectservice.service;

import com.smartfreelance.projectservice.entity.Task;

import java.util.List;

public interface TaskService {
    Task createTask(Task task);
    List<Task> getTasksByPhaseId(Long phaseId);
    Task getTaskById(Long id);
    Task updateTask(Long id, Task task);
    void deleteTask(Long id);
}

File: smartfreelance-frontend/src/app/services/task.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Task } from '../models/task.model';

@Injectable({
  providedIn: 'root'
})
export class TaskService {

  private apiUrl = 'http://localhost:8080/api/tasks';

  constructor(private http: HttpClient) {}

  createTask(task: Task): Observable<Task> {
    return this.http.post<Task>(this.apiUrl, task);
  }

  getTasksByPhaseId(phaseId: number): Observable<Task[]> {
    return this.http.get<Task[]>(`${this.apiUrl}/phase/${phaseId}`);
  }

  updateTask(id: number, task: Task): Observable<Task> {
    return this.http.put<Task>(`${this.apiUrl}/${id}`, task);
  }

  deleteTask(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}